File: ITapply/ITapply.Services/Database/Education.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ITapply.Services.Database
{
    public class Education
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CandidateId { get; set; }
        [ForeignKey("CandidateId")]
        public Candidate Candidate { get; set; }

        [Required]
        [StringLength(200)]
        public string Institution { get; set; }

        [Required]
        [StringLength(100)]
        public string Degree { get; set; } // e.g., Bachelor's, Master's

        [Required]
        [StringLength(100)]
        public string FieldOfStudy { get; set; } // e.g., Computer Science, Electrical Engineering

        [Required]
        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; } // Nullable if currently studying

        [StringLength(2000)]
        public string? Description { get; set; } // Optional, e.g., relevant coursework, GPA
    }
}

File: ITapply/ITapply.UI/itapply_mobile/lib/models/education.dart
import 'package:json_annotation/json_annotation.dart';

part 'education.g.dart';
@JsonSerializable()
class Education {
  final int id;
  final int candidateId;
  final String? candidateName;
  final String institution;
  final String degree;
  final String fieldOfStudy;
  final DateTime startDate;
  final DateTime? endDate;
  final String? description;
  final bool isCurrent;
  final String? duration;

  Education({
    required this.id,
    required this.candidateId,
    this.candidateName,
    required this.institution,
    required this.degree,
    required this.fieldOfStudy,
    required this.startDate,
    this.endDate,
    this.description,
    required this.isCurrent,
    this.duration,
  });

  factory Education.fromJson(Map<String, dynamic> json) => _$EducationFromJson(json);
  Map<String, dynamic> toJson() => _$EducationToJson(this);
}